mmengine/registry/registry.py

```diff
@@ -120,10 +124,25 @@ def __contains__(self, key):
         return self.get(key) is not None
 
     def __repr__(self):
-        format_str = self.__class__.__name__ + \
-                     f'(name={self._name}, ' \
-                     f'items={self._module_dict})'
-        return format_str
+        table = Table(title=f'Registry of {self._name}')
+        table.add_column('Names', justify='right', style='cyan')
+        table.add_column('Objects', justify='right', style='green')
+
+        # In table header, len('Names') == 5 and len('Objects') == 7
+        max_name_width = 5
+        max_obj_width = 7
+        for name, obj in self._module_dict.items():
+            max_name_width = max(max_name_width, len(name))
+            max_obj_width = max(max_obj_width, len(str(obj)))
```

Comment:
If we do not set the `table_width`, can it work as expected?

Reply:
![Snipaste_2023-02-13_17-12-34](https://user-images.githubusercontent.com/44885136/218417399-32f5a6e8-d147-4a83-b94b-cd0daf6e19e6.png)
The `rich` will automatically hide some chars, and broke table struct, which seems not good.

Reply:
If table_width is set, the longest string will be splitter two lines.

![image](https://user-images.githubusercontent.com/58739961/218645509-1869e433-48c0-4248-924f-ede3260fbfac.png)

If table_width is not set, it works well.

![image](https://user-images.githubusercontent.com/58739961/218645734-0a980a64-1420-478d-b4f9-7761a10d86ec.png)

Reply:
@zhouzaida Thank you, you're right. The previous img I attached was “a test unit captured” std output, which might be different from interactive terminal. I'll fix the width issue ASAP.

Reply:
![image](https://user-images.githubusercontent.com/44885136/218655318-954337db-94c5-4453-9a2c-d69fe4452b97.png)
That's how print result looks now. It turns out "setting width" is not necessary :)